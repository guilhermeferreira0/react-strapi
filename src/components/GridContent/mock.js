export default {
  title: 'O titulo que eu quero',
  html: `
  <ul>
    <li>Morbi in sem quis dui placerat ornare. Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu. Cras consequat.</li>
    <li>Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus.</li>
    <li>Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.</li>
    <li>Pellentesque fermentum dolor. Aliquam quam lectus, facilisis auctor, ultrices ut, elementum vulputate, nunc.</li>
  </ul>

  <form action="#" method="post">
    <div>
        <label for="name">Text Input:</label>
        <input type="text" name="name" id="name" value="" tabindex="1" />
    </div>

    <div>
        <h4>Radio Button Choice</h4>

        <label for="radio-choice-1">Choice 1</label>
        <input type="radio" name="radio-choice-1" id="radio-choice-1" tabindex="2" value="choice-1" />

      <label for="radio-choice-2">Choice 2</label>
        <input type="radio" name="radio-choice-2" id="radio-choice-2" tabindex="3" value="choice-2" />
    </div>

    <div>
    <label for="select-choice">Select Dropdown Choice:</label>
    <select name="select-choice" id="select-choice">
      <option value="Choice 1">Choice 1</option>
      <option value="Choice 2">Choice 2</option>
      <option value="Choice 3">Choice 3</option>
    </select>
    </div>

    <div>
    <label for="textarea">Textarea:</label>
    <textarea cols="40" rows="8" name="textarea" id="textarea"></textarea>
    </div>

    <div>
      <label for="checkbox">Checkbox:</label>
    <input type="checkbox" name="checkbox" id="checkbox" />
    </div>

    <div>
      <input type="submit" value="Submit" />
    </div>
  </form>`,
  background: false,
};
