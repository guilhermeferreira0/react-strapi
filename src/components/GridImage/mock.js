export default {
  title: 'My grid image',
  description: 'lorem ipsum dolor',
  grid: [
    {
      altText: 'Teste 1',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 2',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 3',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 4',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 5',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
    {
      altText: 'Teste 6',
      srcImage:
        'https://thypix.com/wp-content/uploads/2018/05/Sommerlandschaft-Bilder-80.jpg',
    },
  ],
  background: false,
};
