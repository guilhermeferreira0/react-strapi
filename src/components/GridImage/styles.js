import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    border-radius: 8px;
  `}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all .5s ease-in-out;
    cursor: pointer;
    border-radius: 8px;

    &:hover {
      transform: scale(1.4) rotate(10deg);
    }
  `}`;
