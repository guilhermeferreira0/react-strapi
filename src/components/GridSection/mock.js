export default {
  title: 'My grid',
  description: 'lorem ipsum dolor',
  grid: [
    {
      title: 'Teste 1',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
    {
      title: 'Teste 2',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
    {
      title: 'Teste 3',
      description:
        'lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds',
    },
  ],
  background: false,
};
