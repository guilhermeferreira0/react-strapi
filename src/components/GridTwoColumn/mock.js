export default {
  title: 'GridTwoColumns',
  text: 'lorem ipsuma sodkaol Ã§lsdp spdlpdlsd sd lrrem lomr lomrem',
  srcImage: 'assets/images/logo.svg',
  background: false,
};
