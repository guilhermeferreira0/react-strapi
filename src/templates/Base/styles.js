import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 5.4rem;
    width: 100%;
    height: 100%;
    overflow: hidden;

    @media (max-width: 975px) {
      padding-top: 8.0rem;
    }

    @media ${theme.media.ltemedium} {
      padding-top: 0;
    }

  `}`;
