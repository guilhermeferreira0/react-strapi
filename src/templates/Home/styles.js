import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${(props) => css`
    background-color: ${props.background};
  `}
`;
