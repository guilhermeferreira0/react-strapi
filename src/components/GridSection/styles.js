import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.ltemedium} {
      grid-template-columns: 1fr;
    }
  `}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 7rem;
    }

    ${HeadingContainer}::before {
      content: counter(grid-counter);
      counter-increment: grid-counter;
      position: absolute;
      font-size: ${theme.fonts.sizes.huge};
      top: -2rem;
      left: -7rem;
      transform: rotate(10deg);
    }
  `}`;
