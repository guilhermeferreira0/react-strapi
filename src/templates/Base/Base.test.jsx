import { screen } from '@testing-library/react';
import { Base } from '.';
import { renderTheme } from '../../styles/renderTheme';
import { mockBase } from './mock';

describe('<Base />', () => {
  it('Should render', () => {
    const { container } = renderTheme(<Base {...mockBase} />);
    expect(container).toMatchSnapshot();
  });
});
