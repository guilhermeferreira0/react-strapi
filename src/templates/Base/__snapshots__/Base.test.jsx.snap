// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Base /> Should render 1`] = `
.c11 {
  padding-top: 5.4rem;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: black;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c14 {
  color: #FFFFFF;
  font-size: 6.2rem;
  text-transform: uppercase;
}

.c19 {
  color: #FFFFFF;
  font-size: 2.4rem;
  text-transform: none;
}

.c21 {
  color: black;
  font-size: 6.2rem;
  text-transform: uppercase;
}

.c22 {
  color: black;
  font-size: 2.4rem;
  text-transform: none;
}

.c2 {
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 14.5%;
  border-bottom: #DDDDDD;
  background: #FFFFFF;
  transition: all 0.6s ease-in-out;
}

.c2 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c2 .c6 {
  margin-top: 0;
  margin-bottom: 0;
}

.c5 {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.c0 {
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 3rem;
  height: 3rem;
  background-color: black;
  color: #FFFFFF;
  border: none;
  display: none;
  cursor: pointer;
  pointer-events: all;
}

.c0 >svg {
  width: 2.5rem;
  height: 2.5rem;
}

.c8 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c8>img {
  max-height: 3rem;
}

.c9 {
  display: flex;
  flex-flow: row wrap;
}

.c10 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: black;
  position: relative;
  scale: 1;
  transition: scale 0.5s ease-in-out;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0%;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 0.5s ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c10:hover {
  scale: 1.1;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c16 {
  font-size: 2.4rem;
}

.c23 {
  text-align: center;
  border-top: 0.1rem solid #DDDDDD;
}

.c23 a {
  color: inherit;
  text-decoration: none;
}

.c23 .c15 {
  font-size: 1.6rem;
}

.c23 .c3 {
  padding-top: 0;
  padding-bottom: 0;
}

.c24 {
  position: fixed;
  background: black;
  color: #FFFFFF;
  border-radius: 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
}

.c13 .c15 {
  margin-bottom: 6.2rem;
}

.c17 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 3.2rem;
}

.c18 .c6 {
  position: relative;
  left: 7rem;
}

.c18 .c6::before {
  content: counter(grid-counter);
  counter-increment: grid-counter;
  position: absolute;
  font-size: 5.6rem;
  top: -2rem;
  left: -7rem;
  transform: rotate(10deg);
}

.c12 {
  background-color: #FFFFFF;
  color: black;
  background-color: black;
  color: #FFFFFF;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c20 {
  background-color: #FFFFFF;
  color: black;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

@media (max-width: 975px) {
  .c11 {
    padding-top: 8.0rem;
  }
}

@media (max-width:768px) {
  .c11 {
    padding-top: 0;
  }
}

@media (max-width:768px) {
  .c14 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c21 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    height: 100vh;
    margin-left: 350px;
    opacity: 0;
  }

  .c2 .c3 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
    overflow-y: auto;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: flex;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c5 {
    display: block;
    text-align: center;
    padding: 4.8rem 0;
  }
}

@media (max-width:768px) {
  .c0 {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c9 {
    flex-flow: column wrap;
    align-content: center;
  }
}

@media (max-width:768px) {
  .c10:hover::after {
    left: 20.5%;
    width: 50%;
  }
}

@media (max-width:768px) {
  .c17 {
    grid-template-columns: 1fr;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <nav
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="c10"
            href="#target1"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </nav>
  <div
    class="c11"
  >
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h3
            class="c6 c14"
          >
            My grid
          </h3>
          <div
            class="c15 c16"
          >
            lorem ipsum dolor
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 1
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 2
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 3
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h3
            class="c6 c21"
          >
            My grid
          </h3>
          <div
            class="c15 c16"
          >
            lorem ipsum dolor
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 1
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 2
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 3
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c12"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h3
            class="c6 c14"
          >
            My grid
          </h3>
          <div
            class="c15 c16"
          >
            lorem ipsum dolor
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 1
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 2
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c19"
              >
                Teste 3
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c20"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c13"
        >
          <h3
            class="c6 c21"
          >
            My grid
          </h3>
          <div
            class="c15 c16"
          >
            lorem ipsum dolor
          </div>
          <div
            class="c17"
          >
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 1
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 2
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
            <div
              class="c18"
            >
              <h1
                class="c6 c22"
              >
                Teste 3
              </h1>
              <div
                class="c15 c16"
              >
                lorem ipsum dolor sit amet consecteturadipisicnt esse ut atque doloreis nobis commoix maxime distinctin veritatus und reprehenderet mineus ad dolores privident maiorsds
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="c23"
    >
      <div
        class="c3 c4"
      >
        <div
          class="c15 c16"
        >
          <p>
            Teste de footer
          </p>
        </div>
      </div>
    </footer>
  </div>
  <a
    aria-label="Go to Top"
    class="c24"
    href="#"
    title="Go to Top"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
